### DISCUSSION TITLE
Backing up and restoring original firmware on Dreamer

### DISCUSSION THREAD

**Discussion by freeifier** [moonglow/FlashForge_Marlin#113] [Q&A]
Hey,

I just stumbled upon your cool project because I am having issues with octoprint and my printer.

I would like to try Marlin but, I really want a backup solution so I can revert to original Firmware.

I found a page where you are using the Jtag connector,
but is that the only option ? 

Cheers

**Comment by moonglow** [comment]
Hi @freeifier !

JTAG allow you make a full backup includes bootloader and so on ( actually it used only by devs or if you got very hard situation without any other ways to restore original firmware )... you can use our python update script under linux or FF Firmware writer ( under windows ). First of all you can write original firmware to your printer with provided tools and if all be OK you can update to Marlin firmware and you be sure that you will restore it back quickly.
